build/wrapper.cc

```diff
@@ -257,62 +257,68 @@ JuliaGcsClient::JuliaGcsClient(const std::string &gcs_address) {
 }
 
 Status JuliaGcsClient::Connect() {
-    gcs_client_ = std::make_unique<gcs::PythonGcsClient>(options_);
-    return gcs_client_->Connect();
+    io_service_ = std::make_unique<instrumented_io_context>();
+    io_service_thread_ = std::make_unique<std::thread>([this] {
+        std::unique_ptr<boost::asio::io_service::work> work(
+            new boost::asio::io_service::work(*io_service_));
+        io_service_->run();
+    });
+    gcs_client_ = std::make_unique<gcs::GcsClient>(options_);
+    return gcs_client_->Connect(*io_service_);
+}
+
+void JuliaGcsClient::Disconnect() {
+    io_service_->stop();
+    io_service_thread_->join();
+    gcs_client_->Disconnect();
+    gcs_client_.reset();
 }
 
-std::string JuliaGcsClient::Get(const std::string &ns,
-                                const std::string &key,
-                                int64_t timeout_ms) {
+std::string JuliaGcsClient::Get(const std::string &ns, const std::string &key) {
     if (!gcs_client_) {
         throw std::runtime_error("GCS client not initialized; did you forget to Connect?");
     }
     std::string value;
-    Status status = gcs_client_->InternalKVGet(ns, key, timeout_ms, value);
+    Status status = gcs_client_->InternalKV().Get(ns, key, value);
     if (!status.ok()) {
         throw std::runtime_error(status.ToString());
     }
     return value;
 }
 
-int JuliaGcsClient::Put(const std::string &ns,
+bool JuliaGcsClient::Put(const std::string &ns,
                         const std::string &key,
                         const std::string &value,
-                        bool overwrite,
-                        int64_t timeout_ms) {
+                        bool overwrite) {
     if (!gcs_client_) {
         throw std::runtime_error("GCS client not initialized; did you forget to Connect?");
     }
-    int added_num;
-    Status status = gcs_client_->InternalKVPut(ns, key, value, overwrite, timeout_ms, added_num);
+    bool added_num;
+    Status status = gcs_client_->InternalKV().Put(ns, key, value, overwrite, added_num);
     if (!status.ok()) {
         throw std::runtime_error(status.ToString());
     }
     return added_num;
 }
 
-std::vector<std::string> JuliaGcsClient::Keys(const std::string &ns,
-                                              const std::string &prefix,
-                                              int64_t timeout_ms) {
+std::vector<std::string> JuliaGcsClient::Keys(const std::string &ns, const std::string &prefix) {
     if (!gcs_client_) {
         throw std::runtime_error("GCS client not initialized; did you forget to Connect?");
     }
     std::vector<std::string> results;
-    Status status = gcs_client_->InternalKVKeys(ns, prefix, timeout_ms, results);
+    Status status = gcs_client_->InternalKV().Keys(ns, prefix, results);
```

Comment:
Looks like `gcs_client_->InternalKV().AsyncInternalKVKeys(...)` may be the thread to pull on for specifying a non-ENV timeout. I noticed `AsyncGetByName` in the `accessor.h` file which does have a timeout. It's unclear why some `Async...` methods have timeouts and other's don't

Reply:
I think it's limited to Actors stuff? Only `AsyncRegisterActor`, `AsyncListNamedActors`, `AsyncGetByName`  have `timeout_ms` as a paramter.

`InternalKVKeys` doesn't: https://github.com/ray-project/ray/blob/a03efd9931128d387649dd48b0e4864b43d3bfb4/src/ray/gcs/gcs_client/accessor.h#L698C1-L698C1

Reply:
The `timeout_ms` is limited to actors but if you did really want it I was saying these `Async` methods would allow you to specify it. For example: https://github.com/ray-project/ray/blob/a03efd9931128d387649dd48b0e4864b43d3bfb4/src/ray/gcs/gcs_client/accessor.cc#L1092-L1110

Reply:
sorry I understand now! I think it's ok to punt for now? we know where to look at least if we want it later?

Reply:
You can definitely punt on it for now. I just wanted to mention this option.